kzg_prover/bin/gen_commit_and_proofs.rs

```diff
@@ -0,0 +1,230 @@
+#![feature(generic_const_exprs)]
+use ethers::types::U256;
+use halo2_proofs::{
+    arithmetic::Field,
+    halo2curves::{
+        bn256::{Bn256, Fr as Fp, G2Affine},
+        group::Curve,
+    },
+    poly::kzg::commitment::KZGCommitmentScheme,
+};
+use num_bigint::BigUint;
+use serde::{Deserialize, Serialize};
+use serde_json::to_string_pretty;
+use std::{fs::File, io::Write};
+use summa_solvency::{
+    circuits::{
+        univariate_grand_sum::UnivariateGrandSum,
+        utils::{full_prover, full_verifier, generate_setup_artifacts},
+    },
+    cryptocurrency::Cryptocurrency,
+    entry::Entry,
+    utils::{
+        amortized_kzg::{commit_kzg, create_naive_kzg_proof, verify_kzg_proof},
+        big_uint_to_fp, parse_csv_to_entries,
+    },
+};
+
+const K: u32 = 17;
+const N_CURRENCIES: usize = 2;
+const N_USERS: usize = 16;
+
+#[derive(Serialize, Deserialize)]
+struct CommitmentSolidityCallData {
+    range_check_snark_proof: String,
+    grand_sums_batch_proof: String,
+    total_balances: Vec<U256>,
+}
+
+#[derive(Serialize, Deserialize)]
+struct InclusionProofCallData {
+    proof: String,
+    challenges: Vec<U256>,
+    username: String,
```

Comment:
Let's rename these fields to "userId" and "userBalances"

Reply:
Yes, that makes sense. Okay, I will replace 'username' with 'userId'

Reply:
It updated at https://github.com/summa-dev/summa-solvency/commit/6db5fb7cc56192741c86d55ff8f59c9eb7dea05b